### ISSUE TITLE
How could it work with argparse?

### ISSUE THREAD

**Issue by alanwilter** [painless-software/python-cli-test-helpers#1]
Could you enrich with an `argparse` example please.

**Comment by bittner** [comment]
The [tests](https://github.com/painless-software/python-cli-test-helpers/tree/main/examples/tests) in the example are agnostic of any CLI framework or module:

1. [test_cli.py](https://github.com/painless-software/python-cli-test-helpers/blob/main/examples/tests/test_cli.py) covers the external behavior of the CLI application by way of brute-force `os.system` calls.
2. [test_command.py](https://github.com/painless-software/python-cli-test-helpers/blob/main/examples/tests/test_command.py) covers the internal behavior of the CLI application, after the CLI framework or module has done its job, by calling Python code directly, unit test-style.

In other words, which CLI framework – if any! – is used in [example/cli.py](https://github.com/painless-software/python-cli-test-helpers/blob/main/examples/foobar/cli.py) is irrelevant to the show case. The implementation is exchangeable in its entirety.

Can you help me understand what you think needs to be done to make the example more valuable?

**P.S.:** The official Python documentation sports an [Argparse Tutorial](https://docs.python.org/3/howto/argparse.html) if that helps.

**Comment by bittner** [comment]
@alanwilter If you're still looking for an argparse example take a look at [bittner/pyclean](https://github.com/bittner/pyclean).

- [argparse-based CLI](https://github.com/bittner/pyclean/blob/main/pyclean/cli.py) (cli.py)
- [related tests](https://github.com/bittner/pyclean/blob/main/tests/test_cli.py) (test_cli.py)

**Comment by alanwilter** [comment]
Thanks a lot @bittner. I had come out with a similar solution but now I must lear about mock and your examples are very helpful.

**Comment by bittner** [comment]
JFYI, the [examples](https://github.com/painless-software/python-cli-test-helpers/tree/main/examples) now sport a dedicated argparse project. :rocket:
